crates/kitsune2/src/factories/mem_peer_store.rs

```diff
@@ -0,0 +1,261 @@
+//! A production-ready memory-based peer store.
```

Comment:
Might be good to have some more docs about the architecture and rationale.

Reply:
Done: https://github.com/holochain/kitsune2/pull/19/commits/44e4bf23e4d2e0b35f5a2bdcac51a5563775b1cb

I put it on the MemPeerStoreFactory because that is the only bit actually exported.